README.md
<br />
<div align="center">
  <a href="https://github.com/Shine-Theme/halo-theme-crux">
    <img src="https://alkaid.ink/upload/crux-banner.png" alt="Banner" width="50%">
  </a>

<h3 align="center">Crux</h3>

  <p align="center">
    A modern minimalist halo theme | 一个现代简约的Halo主题
    <br />
    <a href="https://github.com/Shine-Theme/halo-theme-crux">View Demo</a>
    ·
    <a href="https://docs.alkaid.ink">View Docs</a>
    ·
    <a href="https://github.com/Shine-Theme/halo-theme-crux/issues">Report Bug</a>
    ·
    <a href="https://github.com/Shine-Theme/halo-theme-crux/issues">Request Feature</a>
    <br />
    
[![Stargazers][stars-shield]][stars-url]
[![Issues][issues-shield]][issues-url]
[![MIT License][license-shield]][license-url]
  </p>
</div>

## About The Project

[![Product Name Screen Shot](https://alkaid.ink/upload/crux-screen-shot.png)](https://alkaid.ink)

A modern and minimalist blog theme, suitable for Halo 2.x.Fast iteration with diverse features.Adopting a card-style design solution and responsive design for a better mobile experience.

### Build With.
This project is proudly built on the following technologies.

1. Halo
2. Tailwind CSS
3. alpinejs
4. tocbot
5. iconify-json
6. tailwindcss-plugin-icons

This project proudly utilizes the following design resources.

1. 国内社交媒体图标合集 | by 展哥更新了
2. Octicons | by Github
3. heroicons
4. tablers
5. Unicon | by IconScout
6. Brands logos | By Andrey Kovalev

If there are any outstanding matters, please contact AlkaidMegrez(AlkaidMegrez@outlook.com) for feedback.

## Getting Started

You can go to the Release page to download the latest version, or follow the steps below to compile it yourself.

### Compile

1. Clone the repo
   ```sh
   git clone https://github.com/Shine-Theme/halo-theme-crux.git
   ```
2. Install NPM packages
   ```sh
   pnpm install
   ```
3. Customize Your Theme
   ```sh
   pnpm run dev
   ```
4. Compile your static resources
   ```sh
   pnpm run build
   ```
5. ZIP your file
   ```
   *The following content needs to be included in your ZIP file
   *For reference only.

   .
   ├── templates
   │   ├── modules
   │   │   ├── components
   │   │   │   ├── archives-post-card.html
   │   │   │   ├── footer-basic.html
   │   │   │   ├── footer.html
   │   │   │   ├── health-check.html
   │   │   │   ├── hero.html
   │   │   │   ├── nav.html
   │   │   │   ├── post-card.html
   │   │   │   ├── social-card.html
   │   │   │   ├── social.html
   │   │   │   └── user-card.html
   │   │   ├── icon
   │   │   │   ├── github.html
   │   │   │   ├── links.html //remove soon
   │   │   │   └── qq.html
   │   │   └── layout.html
   │   ├── assets
   │   │   └── dist
   │   │       ├── main.iife.js
   │   │       └── style.css
   │   ├── archives.html
   │   ├── author.html
   │   ├── categories.html
   │   ├── category.html
   │   ├── index.html
   │   ├── links.html
   │   ├── moments.html
   │   ├── page.html
   │   ├── page_about_personal.html
   │   ├── page_github.html
   │   ├── photos.html
   │   ├── post.html
   │   ├── tag.html
   │   └── tags.html
   ├── annotation-setting.yaml
   ├── settings.yaml
   └── theme.yaml
   ```
## Usage

Upload your ZIP file to your blog system (Halo 2.x) and enable your theme, following the documentation or availability testing to perfect the settings.

Then enjoy the fruits of your labor.
## Roadmap

- [x] Responsive design
- [x] Full route adaptation
- [x] Availability testing
- [x] Posts toc
- [ ] Dark Mode(coming soon...)
- [ ] Personalized Page
    - [x] Personal About Page(To be optimized.)
    - [x] Github(To be optimized.)
    - [ ] Team About Page(coming soon...)
    - [ ] Bilibili
    - [ ] Douban

- [ ] `We always on the road`...

See the [open issues](https://github.com/Shine-Theme/halo-theme-crux/issues) for a full list of proposed features (and known issues).


## Contributing

Contributions are what make the open source community such an amazing place to learn, inspire, and create. Any contributions you make are **greatly appreciated**.

If you have a suggestion that would make this better, please fork the repo and create a pull request. You can also simply open an issue with the tag "enhancement".
Don't forget to give the project a star! Thanks again!

## License

```
MIT License

Copyright (c) 2023 Shine Theme Team

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
```

## Contact

AlkaidMegrez - AlkaidMegrez@outlook.com

Project Link: [https://github.com/Shine-Theme/halo-theme-crux](https://github.com/Shine-Theme/halo-theme-crux)

## Special thanks

Thank you to the following organizations or individuals for their strong support during the improvement process of this project.

- [Halo Team](https://github.com/halo-dev)
- [ruibaby](https://github.com/ruibaby)
- [loongzcode](https://github.com/loongzcode)



[contributors-shield]: https://img.shields.io/github/contributors/Shine-Theme/halo-theme-crux.svg?style=for-the-badge
[contributors-url]: https://github.com/Shine-Theme/halo-theme-crux/graphs/contributors
[forks-shield]: https://img.shields.io/github/forks/Shine-Theme/halo-theme-crux.svg?style=for-the-badge
[forks-url]: https://github.com/Shine-Theme/halo-theme-crux/network/members
[stars-shield]: https://img.shields.io/github/stars/Shine-Theme/halo-theme-crux.svg?style=for-the-badge
[stars-url]: https://github.com/Shine-Theme/halo-theme-crux/stargazers
[issues-shield]: https://img.shields.io/github/issues/Shine-Theme/halo-theme-crux.svg?style=for-the-badge
[issues-url]: https://github.com/Shine-Theme/halo-theme-crux/issues
[license-shield]: https://img.shields.io/github/license/Shine-Theme/halo-theme-crux.svg?style=for-the-badge
[license-url]: https://github.com/Shine-Theme/halo-theme-crux/blob/master/LICENSE.txt

CHANGELOG.md
# Changelog

### 1.1.2 (2023-07-15)


### Bug Fixes

* **darkmod:** Fixed Dark Mode when open new page ([3893cc5](https://github.com/Shine-Theme/halo-theme-crux/commit/3893cc58c5d28e3395055df763ffe3715e607a07))
* **nav:** Add a nav title that jumps to the homepage when clicked. ([16140ab](https://github.com/Shine-Theme/halo-theme-crux/commit/16140ab0e844398c9ce8d9c5e03b4ba2a84fe0ed))
* **post:** :bug: fixed the problem where the post could not be previewed. ([d65824b](https://github.com/Shine-Theme/halo-theme-crux/commit/d65824b12d7c201395f04dfb6a4913a09a6af712))

### 1.1.1 (2023-07-11)


### Features

* Add support for JensDll/tailwindcss-plugin-icons ([ea6513a](https://github.com/Shine-Theme/halo-theme-crux/commit/ea6513a1944f37ed071ff4fc86b0803661342c8a))
* **colorScheme:** :sparkles: add support for Dark Mode ([0bd8124](https://github.com/Shine-Theme/halo-theme-crux/commit/0bd8124e8b620f44851974fe2c902592c0b9ba35)), closes [#19](https://github.com/Shine-Theme/halo-theme-crux/issues/19)
* **components:** Add new components ([77d9c24](https://github.com/Shine-Theme/halo-theme-crux/commit/77d9c24eadecbdf693a9364db393afb6299f0a04))
* **config:** Add and modify Halo standard template styles ([67fe29c](https://github.com/Shine-Theme/halo-theme-crux/commit/67fe29c02eda8a92cdf9a94e9b1641ac2861d564))
* **config:** Add new dependencies to npm configuration ([4c9cffb](https://github.com/Shine-Theme/halo-theme-crux/commit/4c9cffb09b0aeb59502f88dd2feb2ede1225ac59))
* **config:** Add new settings options to Halo configuration ([b76b684](https://github.com/Shine-Theme/halo-theme-crux/commit/b76b6840880ed6308d03fbaec6e4c8fa15d55a06))
* **config:** Add support for Halo plugin Photos and Links ([#1](https://github.com/Shine-Theme/halo-theme-crux/issues/1)) ([b080d75](https://github.com/Shine-Theme/halo-theme-crux/commit/b080d755480cdc73cfa70553e3857e233054364b))
* **layout:** Modified the scrollbar style ([8c80ae0](https://github.com/Shine-Theme/halo-theme-crux/commit/8c80ae0fed49ada4a3d4a432d23bade5ac8a570f))
* **post:** Add icons to distinguish the pinned articles([#7](https://github.com/Shine-Theme/halo-theme-crux/issues/7)) ([823b545](https://github.com/Shine-Theme/halo-theme-crux/commit/823b545c4a4ae344b56993c902578c1042660f55))
* **post:** Added post toc ([73f3624](https://github.com/Shine-Theme/halo-theme-crux/commit/73f3624211debbeafd919db54d61ad773ebd083b))
* **theme:** Add availability testing for initialize Crux([#7](https://github.com/Shine-Theme/halo-theme-crux/issues/7)) ([02ae2d6](https://github.com/Shine-Theme/halo-theme-crux/commit/02ae2d672f02d7adb0669bc230032e2da59a8820))


### Bug Fixes

* **config:** Added Theme Logo ([ed5d085](https://github.com/Shine-Theme/halo-theme-crux/commit/ed5d085e6befc362e41ad0f3cce73e9e0e6d13f5))
* **layout:** Deleted useless require ([01e49cf](https://github.com/Shine-Theme/halo-theme-crux/commit/01e49cf278bd1d409d2d832fd12c8ccf234d5b78))
* **moments:** Optimize Moments style ([5238e7d](https://github.com/Shine-Theme/halo-theme-crux/commit/5238e7ddb02a94268c820ae808d814b06f1d1107))
* **nav:** Fix navbar mobile adaptation([#13](https://github.com/Shine-Theme/halo-theme-crux/issues/13)) ([2935c4e](https://github.com/Shine-Theme/halo-theme-crux/commit/2935c4ea59859693e213e6180adf816c9bcfe1dd))
* **page:** Added singlePage temple ([cc7a710](https://github.com/Shine-Theme/halo-theme-crux/commit/cc7a7101281cfb6f110e587bd1ba139a45b2322f))
* **security:** Resolve accidental GitHub Personal Token leak ([5f17f07](https://github.com/Shine-Theme/halo-theme-crux/commit/5f17f071394cec6156a77ee532967e2e51f7ea39))
* **toc:** Added script ([3b12135](https://github.com/Shine-Theme/halo-theme-crux/commit/3b12135b08cde0439a53ffd75ce72648a78179aa))
* **toc:** Fixed toc in mobile ([5a170cc](https://github.com/Shine-Theme/halo-theme-crux/commit/5a170cc9908c37ec10468699100d4ab4e22cb64d))
* **toc:** Use Alpinejs to generate toc ([8a57e1b](https://github.com/Shine-Theme/halo-theme-crux/commit/8a57e1b1d4213c644a2908e52d4b79a720728f60))


src/data/toc.ts
import * as tocbot from "tocbot";
export default () => ({
  generateToc(content_id: string, toc_container_class: string, toc_class: string) {
    const content = document.getElementById(content_id);
    const titles = content?.querySelectorAll("h1, h2, h3, h4");
    if (!titles || titles.length === 0) {
      const tocContainer = document.querySelector("." + toc_container_class);
      tocContainer?.remove();
      return;
    }
    tocbot.init({
      tocSelector: "." + toc_class,
      contentSelector: "#" + content_id,
      headingSelector: "h1, h2, h3, h4",
      extraListClasses: "space-y-1 dark:border-slate-500",
      extraLinkClasses:
        "group flex items-center justify-between rounded py-1 px-1.5 transition-all hover:bg-gray-100 text-sm opacity-80 dark:hover:bg-slate-700 dark:text-slate-50",
      activeLinkClass: "is-active-link",
      collapseDepth: 6,
      headingsOffset: 100,
      scrollSmooth: true,
      scrollSmoothOffset: -120,
    });
  },

  // this function copy from https://github.com/halo-dev/theme-earth/blob/main/src/main.ts
});


src/data/github.ts
/*
 * @Author: Alkaid(AlkaidMegrez@outlook.com)
 * @Date: 2023-07-11 00:08:17
 * @LastEditors: Alkaid(AlkaidMegrez@outlook.com)
 * @LastEditTime: 2023-07-11 18:00:13
 * @FilePath: /theme-crux/src/data/github.ts
 * @Description: 当前解决方案会泄漏github_token，不建议使用，即将弃用
 *
 * Copyright (c) 2023 by Crux, All Rights Reserved.
 */

import axios from "axios";
export default () => ({
  loading: true,
  error: false,
  userinfo: {},
  repos: {},
  get(github_username: string, github_token: string) {
    axios
      .get("https://api.github.com/users/" + github_username, {
        headers: {
          Authorization: "token " + github_token,
        },
      })
      .then((response) => {
        this.userinfo = response.data;
        axios
          .get(response.data.repos_url, {
            headers: {
              Authorization: "token " + github_token,
            },
          })
          .then((response) => {
            this.repos = response.data;
            this.loading = false;
          })
          .catch(() => {
            this.loading = false;
            this.error = true;
          });
      })
      .catch(() => {
        this.loading = false;
        this.error = true;
      });
  },
});


src/data/darkmod.ts
/*
 * @Author: Alkaid(AlkaidMegrez@outlook.com)
 * @Date: 2023-07-11 14:52:45
 * @LastEditors: Alkaid(AlkaidMegrez@outlook.com)
 * @LastEditTime: 2023-07-15 21:56:26
 * @FilePath: /theme-crux/src/data/darkmod.ts
 * @Description:
 *
 * Copyright (c) 2023 by Crux, All Rights Reserved.
 */
export default () => ({
  colorscheme: "light",
  init() {
    if (localStorage.getItem("color-scheme") === "dark") {
      this.colorscheme = "dark";
    } else {
      this.colorscheme = "light";
    }
  },
  toggle() {
    if (this.colorscheme == "light") {
      this.colorscheme = "dark";
      localStorage.setItem("color-scheme", "dark");
      document.getElementById("app")?.classList.add("dark");
    } else {
      this.colorscheme = "light";
      localStorage.setItem("color-scheme", "light");
      document.getElementById("app")?.classList.remove("dark");
    }
    console.log(this.colorscheme);
  },
});


tailwind.config.js
/*
 * @Author: Alkaid(AlkaidMegrez@outlook.com)
 * @Date: 2023-07-07 18:00:15
 * @LastEditors: Alkaid(AlkaidMegrez@outlook.com)
 * @LastEditTime: 2023-07-16 04:35:52
 * @FilePath: /theme-crux/tailwind.config.js
 * @Description: 
 * 
 * Copyright (c) 2023 by Crux, All Rights Reserved. 
 */
/** @type {import('tailwindcss').Config} */

const { Icons } = require("tailwindcss-plugin-icons");
module.exports = {
  content: ["./templates/**/*.html", "./src/main.ts"],
  darkMode: ['class'],
  theme: {
    extend: {},
    fontFamily: {
      sans: [
        'ui-sans-serif',
        'system-ui',
        '-apple-system',
        'BlinkMacSystemFont',
        '"Segoe UI"',
        'Roboto',
        '"Helvetica Neue"',
        'Arial',
        '"Noto Sans"',
        'sans-serif',
        '"Apple Color Emoji"',
        '"Segoe UI Emoji"',
        '"Segoe UI Symbol"',
        '"Noto Color Emoji"',
      ],
      serif: ['ui-serif', 'Georgia', 'Cambria', '"Times New Roman"', 'Times', 'serif'],
      mono: [
        'ui-monospace',
        'SFMono-Regular',
        'Menlo',
        'Monaco',
        'Consolas',
        '"Liberation Mono"',
        '"Courier New"',
        'monospace',
      ],
      noto_serif: ["Noto Serif SC"],
    }
  },
  plugins: [
    require("@tailwindcss/typography"),
    Icons(() => ({
      tabler: {
        includeAll: true,
      },
      heroicons: {
        includeAll: true,
      },
      uim: {

        includeAll: true,
      }
    })),
  ],
  safelist: [
    "prose-sm",
    "prose-base",
    "prose-lg",
    "prose-xl",
    "prose-2xl",
    "prose-gray",
    "prose-slate",
    "prose-zinc",
    "prose-neutral",
    "prose-stone",
  ],
};
